File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        l=0
        r=1
        maxProfit=0
        nums=prices
        while r<len(nums):
            curProfit = nums[r]-nums[l]
            maxProfit = max(maxProfit , curProfit)
            if nums[l]>nums[r]:
                l=r
            r=r+1
        return maxProfit

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        int l = 0;
        int maxP=0;
        int curP=0;
        
        for( int r=1;r<prices.size();r++ )
        {
            
            curP= prices[r]-prices[l];
            maxP=max(curP, maxP);
            if (prices[l]>prices[r])
            {
                l=r;
            }
            
        }
        
        return maxP;
        
    }
};